lib/jmap/core/utc_date.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:equatable/equatable.dart';
+
+class UTCDate with EquatableMixin {
+  final String value;
```

Comment:
> Where UTCDate is given as a type, it means a Date where the time-offset component MUST be Z (i.e., it must be in UTC time). For example, "2014-10-30T06:12:00Z".

I think it should be `Date` instead of `String`, but we need a constraint with the offset `Z`